test_conformance/vulkan/vulkan_interop_common/vulkan_utility.cpp

```diff
@@ -0,0 +1,700 @@
+         throw std::runtime_error("failed to find a suitable GPU!");
+    }
+    std::cout << "Selected physical device is: " << physicalDeviceList[pdIdx] << std::endl;
+    return physicalDeviceList[pdIdx];
+}
+
+const VulkanQueueFamily &
+getVulkanQueueFamily(
+    uint32_t queueFlags)
+{
+    size_t qfIdx;
+    const VulkanPhysicalDevice & physicalDevice = getVulkanPhysicalDevice();
+    const VulkanQueueFamilyList & queueFamilyList = physicalDevice.getQueueFamilyList();
+
+    for (qfIdx = 0; qfIdx < queueFamilyList.size(); qfIdx++) {
+        if ((queueFamilyList[qfIdx].getQueueFlags() & queueFlags) == queueFlags) {
+            break;
+        }
+    }
+
+    return queueFamilyList[qfIdx];
+}
+
+const VulkanMemoryType &
+getVulkanMemoryType(
+    const VulkanDevice & device,
+    VulkanMemoryTypeProperty memoryTypeProperty)
+{
+    size_t mtIdx;
+    const VulkanMemoryTypeList & memoryTypeList = device.getPhysicalDevice().getMemoryTypeList();
+
+    for (mtIdx = 0; mtIdx < memoryTypeList.size(); mtIdx++) {
+        if ((memoryTypeList[mtIdx].getMemoryTypeProperty() & memoryTypeProperty) == memoryTypeProperty) {
+            break;
+        }
+    }
+
+   // CHECK_LT(mtIdx, memoryTypeList.size());
+    return memoryTypeList[mtIdx];
+}
+
+const VulkanQueueFamilyList &
+getEmptyVulkanQueueFamilyList()
+{
+    static VulkanQueueFamilyList queueFamilyList;
+
+    return queueFamilyList;
+}
+
+const VulkanDescriptorSetLayoutList &
+getEmptyVulkanDescriptorSetLayoutList()
+{
+    static VulkanDescriptorSetLayoutList descriptorSetLayoutList;
+
+    return descriptorSetLayoutList;
+}
+
+const VulkanQueueFamilyToQueueCountMap &
+getDefaultVulkanQueueFamilyToQueueCountMap()
+{
+    static VulkanQueueFamilyToQueueCountMap queueFamilyToQueueCountMap(1);
+
+    return queueFamilyToQueueCountMap;
+}
+
+const std::vector<VulkanExternalMemoryHandleType>
+getSupportedVulkanExternalMemoryHandleTypeList()
+{
+    std::vector<VulkanExternalMemoryHandleType> externalMemoryHandleTypeList;
+
+#if _WIN32 
+    if (IsWindows8OrGreater()) {
+        externalMemoryHandleTypeList.push_back(VULKAN_EXTERNAL_MEMORY_HANDLE_TYPE_OPAQUE_WIN32_NT);
+    }
+    externalMemoryHandleTypeList.push_back(VULKAN_EXTERNAL_MEMORY_HANDLE_TYPE_OPAQUE_WIN32_KMT);
+#else
+    externalMemoryHandleTypeList.push_back(VULKAN_EXTERNAL_MEMORY_HANDLE_TYPE_OPAQUE_FD);
+#endif
+
+    return externalMemoryHandleTypeList;
+}
+
+const std::vector<VulkanExternalSemaphoreHandleType>
+getSupportedVulkanExternalSemaphoreHandleTypeList()
+{
+    std::vector<VulkanExternalSemaphoreHandleType> externalSemaphoreHandleTypeList;
+
+#if _WIN32 
+    if (IsWindows8OrGreater()) {
+        externalSemaphoreHandleTypeList.push_back(VULKAN_EXTERNAL_SEMAPHORE_HANDLE_TYPE_OPAQUE_WIN32_NT);
+    }
+    externalSemaphoreHandleTypeList.push_back(VULKAN_EXTERNAL_SEMAPHORE_HANDLE_TYPE_OPAQUE_WIN32_KMT);
+#else
+    externalSemaphoreHandleTypeList.push_back(VULKAN_EXTERNAL_SEMAPHORE_HANDLE_TYPE_OPAQUE_FD);
```

Comment:
Can we change this to allow the tests to run on platforms that only support SYNC_FD?

Reply:
We currently don't have SYNC_FD support in this PR. But, we can allow only SYNC_FD when we add support for SYNC_FD.